src/core/contrib.js

```diff
@@ -10,107 +10,86 @@ import { pub } from "core/pubsubhub";
-          return editors.indexOf(name) < 0;
-        });
-      names.sort(function(a, b) {
-        return a.toLowerCase().localeCompare(b.toLowerCase());
-      });
-      $(element).html(join(names)).attr("id", null);
-    });
+async function toHTML(urls, editors, element) {
+  const args = await Promise.all(...urls.map(ghFetch));
+  const names = args
+    .map(user => user[0].name || user[0].login)
+    .filter(name => !editors.includes(name))
+    .sort((a, b) => a.toLowerCase().localeCompare(b.toLowerCase()));
+  element.textContent = join(names);
+  element.id = null;
 }
 
-export async function run(conf, doc, cb) {
-  var $commenters = doc.querySelector("#gh-commenters");
-  var $contributors = doc.querySelector("#gh-contributors");
+export async function run(conf) {
+  const ghCommenters = document.getElementById("gh-commenters");
+  const ghContributors = document.getElementById("gh-contributors");
 
-  if (!$commenters && !$contributors) {
-    return cb();
+  if (!ghCommenters && !ghContributors) {
+    return;
   }
 
   if (!conf.githubAPI) {
-    var elements = [];
-    if ($commenters) elements.push("#" + $commenters.id);
-    if ($contributors) elements.push("#" + $contributors.id);
+    const elements = [];
+    if (ghCommenters) elements.push("#" + ghCommenters.id);
+    if (ghContributors) elements.push("#" + ghContributors.id);
     pub(
       "error",
-      "Requested list of contributors and/or commenters from GitHub (" +
-        elements.join(" and ") +
-        ") but config.githubAPI is not set."
+      `Requested list of contributors and/or commenters from GitHub (${
+        elements.join(" and ")
+      }) but config.githubAPI is not set.`
     );
-    cb();
     return;
   }
 
-  ghFetch(conf.githubAPI)
-    .then(function(json) {
-      return $.when(
-        fetchIndex(json.issues_url),
-        fetchIndex(json.issue_comment_url),
-        fetchIndex(json.contributors_url)
-      );
-    })
-    .then(function(issues, comments, contributors) {
-      var editors = respecConfig.editors.map(prop("name"));
-      var commenters = findUsers(issues, comments);
-      contributors = contributors.map(prop("url"));
-      return $.when(
-        toHTML(commenters, editors, $commenters),
-        toHTML(contributors, editors, $contributors)
-      );
-    })
-    .then(cb, function(error) {
-      pub(
-        "error",
-        "Error loading contributors and/or commenters from  Error: " + error
-      );
-      cb();
-    });
+  const json = await ghFetch(conf.githubAPI);
+  const [issues, comments, contributors] = await Promise.all(
+    fetchIndex(json.issues_url),
+    fetchIndex(json.issue_comment_url),
+    fetchIndex(json.contributors_url)
+  );
+  const editors = respecConfig.editors.map(prop("name"));
+  const commenterUrls = findUsers(issues, comments);
+  const contributorUrls = contributors.map(prop("url"));
+  try {
+    return Promise.all(
+      toHTML(commenterUrls, editors, ghCommenters),
```

Comment:
I’m not sure we want to return the result of Promise.all here. Shouldn’t this await first the return void instead? Alternatively, we could return a promise that resolves with undefined once Promise.all() resolves.